### ISSUE TITLE
INTERNAL_SERVER_ERROR when using @cypher directive

### ISSUE THREAD

**Issue by nsenoge63** [neo4j/graphql#5614]
### Type definitions

```GraphQL
type Composant {
    name: String!
    mail: String!

    "Liste de toutes les versions de ce composant"
    versions: [Version!]! @relationship(type: "EST_LIEE_AU", direction: IN)
}

type Version {
    version: String!
    mail: String @cypher(
        statement: """
        OPTIONAL MATCH (this)-[:EST_LIEE_AU]->(c:Composant)
        RETURN c.mail AS mail
        """,
        columnName: "mail"
   )

    "Composant rattaché à cette version"
    composant: Composant! @relationship(type: "EST_LIEE_AU", direction: OUT)
}
```

### Test data

```Shell
CREATE (c:Composant{name:"composant-1", mail:"myemail@mail.com"})
CREATE (v:Version{version:"1.0.0"})
WITH c, v
CREATE (v)-[:EST_LIEE_AU]->(c)
```

### Steps to reproduce

1. Start server
2. Execute a request

### What happened

When I execute the following query : 

```
query {
  composants {
    versions {
      mail
    }
  }
}
```

The output is ok : 

```
{
  "data": {
    "composants": [
      {
        "versions": [
          {
            "mail": "myemail@mail.com"
          }
        ]
      }
    ]
  }
}
```

However, if I execute a simplier request, I get an INTERNAL_SERVER_ERROR : 

```
query {
  versions {
    mail
  }
}
```

### Expected behaviour

I expected this response : 

```
{
  "data": {
        "versions": [
          {
            "mail": "myemail@mail.com"
          }
        ]
  }
}
```

and not an internal server error

### Version

5.7.1

### Database version

4.4.4

### Relevant log output

```Shell
{
  "errors": [
    {
      "message": "key not found: SymbolUse(this@84(1126149525))",
      "locations": [
        {
          "line": 10,
          "column": 3
        }
      ],
      "path": [
        "versions"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "stacktrace": [
          "Neo4jError: key not found: SymbolUse(this@84(1126149525))",
          ": ",
          "    at captureStacktrace (D:\\DEVOPS\\workspace-intellij\\DATAWAREHOUSE\\graphqlappli\\node_modules\\neo4j-driver-core\\lib\\result.js:624:17)",
          "    at new Result (D:\\DEVOPS\\workspace-intellij\\DATAWAREHOUSE\\graphqlappli\\node_modules\\neo4j-driver-core\\lib\\result.js:112:23)",
          "    at newCompletedResult (D:\\DEVOPS\\workspace-intellij\\DATAWAREHOUSE\\graphqlappli\\node_modules\\neo4j-driver-core\\lib\\transaction.js:528:12)",
          "    at Object.run (D:\\DEVOPS\\workspace-intellij\\DATAWAREHOUSE\\graphqlappli\\node_modules\\neo4j-driver-core\\lib\\transaction.js:360:20)",
          "    at TransactionPromise.Transaction.run (D:\\DEVOPS\\workspace-intellij\\DATAWAREHOUSE\\graphqlappli\\node_modules\\neo4j-driver-core\\lib\\transaction.js:181:34)",
          "    at ManagedTransaction.run (D:\\DEVOPS\\workspace-intellij\\DATAWAREHOUSE\\graphqlappli\\node_modules\\neo4j-driver-core\\lib\\transaction-managed.js:54:21)",
          "    at Executor.transactionRun (D:\\DEVOPS\\workspace-intellij\\DATAWAREHOUSE\\graphqlappli\\node_modules\\@neo4j\\graphql\\src\\classes\\Executor.ts:297:28)",
          "    at D:\\DEVOPS\\workspace-intellij\\DATAWAREHOUSE\\graphqlappli\\node_modules\\@neo4j\\graphql\\src\\classes\\Executor.ts:269:33",
          "    at TransactionExecutor._safeExecuteTransactionWork (D:\\DEVOPS\\workspace-intellij\\DATAWAREHOUSE\\graphqlappli\\node_modules\\neo4j-driver-core\\lib\\internal\\transaction-executor.js:211:26)",
          "    at TransactionExecutor.<anonymous> (D:\\DEVOPS\\workspace-intellij\\DATAWAREHOUSE\\graphqlappli\\node_modules\\neo4j-driver-core\\lib\\internal\\transaction-executor.js:198:46)",
          "    at step (D:\\DEVOPS\\workspace-intellij\\DATAWAREHOUSE\\graphqlappli\\node_modules\\neo4j-driver-core\\lib\\internal\\transaction-executor.js:44:23)",
          "    at Object.next (D:\\DEVOPS\\workspace-intellij\\DATAWAREHOUSE\\graphqlappli\\node_modules\\neo4j-driver-core\\lib\\internal\\transaction-executor.js:25:53)",
          "    at fulfilled (D:\\DEVOPS\\workspace-intellij\\DATAWAREHOUSE\\graphqlappli\\node_modules\\neo4j-driver-core\\lib\\internal\\transaction-executor.js:16:58)",
          "    at processTicksAndRejections (node:internal/process/task_queues:96:5)"
        ]
      }
    }
  ],
  "data": null
}
```

**Comment by mjfwebb** [comment]
Confirmed the error does appear on Neo4j version 4.4.4 and 4.4.5.

Also confirmed that the error no longer appears on 4.4.37.

I'm going to try to track down which version it was fixed in, but ultimately this will require a DB version upgrade @nsenoge63 .

**Comment by mjfwebb** [comment]
It looks like it was fixed in 4.4.6. I do recommend using the latest version though to get all the improvements, bug fixes etc.

Release notes for 4.4.6 if you're interested: https://neo4j.com/release-notes/database/neo4j-4-4-6/

**Comment by nsenoge63** [comment]
> It looks like it was fixed in 4.4.6. I do recommend using the latest version though to get all the improvements, bug fixes etc.
> 
> Release notes for 4.4.6 if you're interested: https://neo4j.com/release-notes/database/neo4j-4-4-6/

Thanks for your feedback, I'll try to upgrade my database
